hooks/useRailgunTx.tsx

```diff
@@ -0,0 +1,188 @@
+  RailgunERC20AmountRecipient,
+  RailgunERC20Amount,
+  NetworkName,
+  TransactionGasDetailsSerialized,
+  EVMGasType,
+  NETWORK_CONFIG,
+  deserializeTransaction,
+} from "@railgun-community/shared-models";
+import { ethers } from "ethers";
+import { getAddress, parseUnits } from "ethers/lib/utils.js";
+import { useAccount, useSigner } from "wagmi";
+import { useProvider } from "wagmi";
+
+const useRailgunTx = () => {
+  const { data: signer } = useSigner();
+  const { address } = useAccount();
+  const { getShieldPrivateKey } = useShieldPrivateKey();
+  const network = NetworkName.EthereumGoerli;
+  const { weth } = useToken();
+  const provider = useProvider();
+
+  const shield = async (args: {
+    tokenAddress: string;
+    tokenAmount: string;
+    tokenDecimals: number;
+    recipient: string;
+  }) => {
+    if (args.tokenAddress === getAddress(`0x${"e".repeat(40)}`))
+      return shieldBaseToken(args);
+    return shieldToken(args);
+  };
+
+  const shieldBaseToken = async ({
+    tokenAmount,
+    tokenDecimals,
+    recipient,
+  }: {
+    tokenAddress: string;
+    tokenAmount: string;
+    tokenDecimals: number;
+    recipient: string;
+  }) => {
+    // Public wallet to shield from.
+    const fromWalletAddress = address as `0x{string}`;
+
+    // The shieldPrivateKey enables the sender to decrypt
+    // the receiver's address in the future.
+    const shieldPrivateKey = await getShieldPrivateKey();
+
+    const wrappedERC20Amount: RailgunERC20Amount = {
+      tokenAddress: weth!, // wETH
+      amountString: parseUnits(tokenAmount!, tokenDecimals).toHexString(), // hexadecimal amount
+    };
+
+    const { gasEstimateString, error: err } =
+      await gasEstimateForShieldBaseToken(
+        network,
+        recipient,
+        shieldPrivateKey,
+        wrappedERC20Amount,
+        fromWalletAddress
+      );
+    if (err) {
+      throw err;
+    }
+
+    const { maxFeePerGas, maxPriorityFeePerGas } = await provider.getFeeData();
+
+    const gasDetailsSerialized: TransactionGasDetailsSerialized = {
+      evmGasType: EVMGasType.Type2, // Depends on the chain (BNB uses type 0)
+      gasEstimateString: gasEstimateString!, // Output from gasEstimateForShieldBaseToken
+      maxFeePerGasString: maxFeePerGas!.toHexString(), // Current gas Max Fee
+      maxPriorityFeePerGasString: maxPriorityFeePerGas!.toHexString(), // Current gas Max Priority Fee
+    };
+
+    const { serializedTransaction, error } = await populateShieldBaseToken(
+      network,
+      recipient,
+      shieldPrivateKey,
+      wrappedERC20Amount,
+      gasDetailsSerialized
+    );
+    if (error) {
+      throw error;
+    }
+
+    const { chain } = NETWORK_CONFIG[network];
+
+    const transactionRequest: ethers.providers.TransactionRequest =
+      deserializeTransaction(
+        serializedTransaction!,
+        undefined, // nonce (optional)
+        chain.id
+      );
+
+    // Public wallet to shield from.
+    transactionRequest.from = address;
```

Comment:
Another SDK comment that we discussed—this line should not be necessary, we should just pass this address into `deserializeTransaction` and get the final `transactionRequest` object back. (Leaving these SDK comments here to see if you agree, then we can port them over to our doc)

Reply:
yeah feels a lil neater